### ISSUE TITLE
Error retrieving URL. Unable to retrieve content. Remote peer disconnected.

### ISSUE THREAD

**Issue by farazirfan47** [cubiclesoft/ultimate-web-scraper#4]
I am getting this error on some URLs. How can I solve this error?

**Comment by cubiclesoft** [comment]
Got an example URL that I can use to consistently reproduce the issue with?

You can also try passing `"debug" => true` in the $options array to see how far you are getting.  For example, does it send but then disconnect or does it just disconnect immediately after connecting.

Also, are you using the latest GitHub version or the RC15 build on the website?  Make sure the GitHub version has the same issue.  1.0RC15 was released on April 18, 2016.  Some things have changed since.

**Comment by cubiclesoft** [comment]
@farazirfan47 Closing this due to the issue's age.  Open a new issue and reference this issue if you still encounter the problem.  Thanks.

**Comment by ablionit** [comment]
not working with 
https://www1.nseindia.com/education/content/reports/eq_research_reports_listed.htm
Error retrieving URL. Unable to retrieve content. Remote peer disconnected.

any help?

**Comment by cubiclesoft** [comment]
Confirmed.  With debugging enabled in the library, I get:

```
PHP Warning:  fread(): SSL: An existing connection was forcibly closed by the remote host.
 in D:\WEB\barebones\ultimate-web-scraper\support\http.php on line 628
```

Looks like that server does User-Agent detection + requires `Connection: keep-alive` instead of `Connection: close`.  It doesn't like the older Firefox 38 User-Agent string.  I've updated the repo to use a Firefox 77 User-Agent and it seems to work when `Connection: keep-alive` is passed in the headers.
